### ISSUE TITLE
Add views for EBS volumes

### ISSUE THREAD

**Issue by cjh1** [Kitware/HPCCloud#5]
Currently the only configurable parameter to expose to the user is the size. 

The REST endpoint can be found here: http://localhost:8080/api/v1#!/volumes

**Comment by cjh1** [comment]
When a cluster is specified we will also need a way for user to attach one of their volumes to the cluster.

**Comment by TristanWright** [comment]
This is in place on the amazon tab, there may be a small problem when deleting the instances the volumes don't detach, so this is maybe 50%

**Comment by cjh1** [comment]
Once Kitware/cumulus#226 lands we will want to add various views and associations for EBS volumes. I am going to use this issue to document them. 

### Volume endpoints

The backend will support the following endpoints:

```
GET /volumes - Search for volumes
POST /volumes - Create a volume
GET /volumes/{id} - Get a volume
DELETE /volumes/{id} - Delete a volume
PUT /volumes/{id}/clusters/{clusterId}/attach - Attach a volume to a cluster
PUT /volumes/{id}/detach - Detach a volume from a cluster
GET /volumes/{id}/status - Get the status of a volume
``` 

### Associations

We will want to present the following association in the UI:

- Cluster can have multiple attached volumes. ( the cluster model contain a list containing the ids of the attached volumes).
- A volume can be attached to a cluster ( the clusterId is saved in the volume model ).
- A volume can contain the results of multiple simulations. We probably want to save the volume ids in the simulation metadata.

### New UI components

#### Associating a volume with a cluster
When a cluster is created a user should be able to attach an existing volume or provide details of a new volume they would like to create for the cluster. Here is a mock up of what this could look like:

WIP

**Comment by chetnieter** [comment]
Can a volume be attached to more than one cluster at a time? If so does 'PUT /volumes/{id}/detach' detach the volume from all clusters?

**Comment by cjh1** [comment]
@TristanWright in @chetnieter absence do you think you could take over this, I would really like to make some progress on it? @chetnieter do you have anything that should be pushed that @TristanWright can build on?

**Comment by chetnieter** [comment]
I started a branch [volumes-view](https://github.com/Kitware/HPCCloud/tree/volumes-view) but really all I had done is copy another file to start working on things. Not sure there is anything there of importance. I don't have anything local that hasn't been pushed to github already. I am guessing I may be back on HPCCloud next week but I there are several other things I need to tackle - the pvbatch stuff and NWChem work - before I can work on this.

**Comment by TristanWright** [comment]
If you do come back it Chet, I stubbed out the volumes page and reducers on top of that branch. I'll pick it up more soon otherwise.

**Comment by TristanWright** [comment]
Are there other type options for volumes besides ebs?

https://github.com/Kitware/cumulus/blob/efe91d4020da33d0076c5f0464bd26fee6d126e2/cumulus/constants.py#L38

A diagram above offers selection between magnetic and ssd etc?

**Comment by cjh1** [comment]
@TristanWright Yes, there will be other volume types, for example OpenStack has its own specific volume types. Which diagram ?

**Comment by cjh1** [comment]
@TristanWright I see the diagram you are referring to now.  I think the volume type field will need a little reworking. We may have to have two fields here. One should indicate the cloud provider (AWS, OpenStack etc.) and the other should indicate the specific type within that provider for example in the case of AWS magnetic, gp2,. io1 etc. I will raise an issue to resolve this, is this blocking you?

**Comment by TristanWright** [comment]
closed with #549
